hammerblade/torch/kernel/bsg_tensor.hpp

```diff
@@ -13,4 +31,100 @@ typedef struct {
 #endif
 } bsg_tensor_t;
 
+typedef struct {
+  uint32_t N;
+#ifdef HB_EMUL
+  uint64_t data;
+#else
+  uint32_t data;
+#endif
+} bsg_vector_t;
+
+// =========================================================
+// Device Tensor classes
+//
+// Wrapper classes around device tensor structs to provide
+// convenience operations. This runs on a tiny RISC-V processor 
+// on the device, so be careful about using dynamic memory 
+// allocation.
+// =========================================================
+
+class BSGTensor {
```

Comment:
I really like the idea of helper classes!

Reply:
Yes, with operator overloading, writing device kernels feels like writing numpy!

Reply:
Definitely! Developing these kind of C++ wrappers will really help kernel evelopment.